src/paperbot/infrastructure/swarm/paper_slug.py

```
@@ -0,0 +1,45 @@
+        → "attention-is-all-you-need-a9b1"
+
+        paper_slug("def456", "")
+        → "paper-d8e2"
+    """
+    if title:
+        readable = re.sub(r"[^a-z0-9]+", "-", title.lower()).strip("-")
+        readable = re.sub(r"-{2,}", "-", readable)
+    else:
+        readable = "paper"
+
+    suffix = hashlib.md5(paper_id.encode()).hexdigest()[:4]
```

Ruff (0.15.5)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)